tokio-buf/Cargo.toml

```diff
@@ -24,11 +24,11 @@ publish = false
 [dependencies]
 bytes = "0.4.10"
 either = { version =  "1.5", optional = true}
-futures = "0.1.23"
+tokio-futures = { path = "../tokio-futures" }
```

Comment:
I would add an optional dependency on futures-core and use a `async-traits` feature flag to guard it.